src/components/search/AggregateSearchResult/index.js

```diff
@@ -7,17 +7,24 @@ module.exports = React.createClass({
 
   propTypes: {
     url: React.PropTypes.string,
+    result: React.PropTypes.object,
     onSelect: React.PropTypes.func
   },
 
+  handleClick: function(e) {
+    e.preventDefault();
+    this.props.onSelect(this.props.result);
+  },
+
   render: function() {
-    var props = this.props;
+    var props       = this.props;
+    var handleClick = props.onSelect && this.handleClick;
```

Comment:
Is the idea here that if you pass a falsey value to the `onClick` prop then React won't attach a listener? I think, seeing as React has one listener and uses delegation, we might not need to worry. Just curious, don't know a heap about React events.